packages/main/src/plugin/kubernetes/contexts-manager.spec.ts

```diff
@@ -38,6 +38,44 @@ const DEPLOYMENTS_DEFAULT = 6;
 const NODES_CONTEXT1 = 1;
 const NODES_CONTEXT2 = 2;
 
+// Adds an asymmetric  matcher
+// https://vitest.dev/guide/extending-matchers
+interface CustomMatchers<R = unknown> {
+  stateMapContaining(a: string, b: CheckingState): R;
+}
+
+declare module 'vitest' {
+  // eslint-disable-next-line @typescript-eslint/no-empty-object-type
+  interface AsymmetricMatchersContaining extends CustomMatchers {}
+}
+
+expect.extend({
+  stateMapContaining(
+    map: Map<string, CheckingState>,
+    key: string,
+    value: CheckingState,
+  ): { message: () => string; pass: boolean } {
+    try {
+      if (this.equals(map.get(key), value)) {
+        return {
+          message: () => `map ${map} contains ${key} with value ${value}`,
+          pass: true,
+        };
+      } else {
+        return {
+          message: () => `map ${map} does not contain ${key} with value ${value}`,
+          pass: false,
+        };
+      }
+    } catch {
+      return {
+        message: () => `${map} is not a map`,
+        pass: false,
+      };
+    }
+  },
+});
+
```

Comment:
here I don't see the value of augmenting expect rather than just having a method getState() method returning true or false

Reply:
We want to test that the checking state has been set to true during the period, but we cannot test that it is currently true, as it can have been changed to false during the period.